### ISSUE TITLE
s/and fails in some cases

### ISSUE THREAD

**Issue by mping** [metosin/spec-tools#165]
Using `s/and` with a single spec, the behaviour should be the same as without s/and (at least thats the principle of least surprise).

```
(:require [clj-time.coerce :as coerce])
(:import  [org.joda.time LocalDate])

(s/def ::custom-date (st/spec {:spec        #(instance? LocalDate %)
                                                    :decode/json #(coerce/to-local-date %2)}))
(st/coerce ::custom-date "2019-01-01" st/json-transformer) ;;#object[org.joda.time.LocalDate 0x7b3e48 "2019-01-01"]
(st/coerce (s/and ::custom-date) "2019-01-01" st/json-transformer) ;;"2019-01-01"

```

**Comment by ikitommi** [comment]
That's not good. It seems that the form-walker just tries to walk over `:spec-tools.parse/items`, which only uses `:type` for the transformation, not the `encode` & `decode` keys. Same applies for all specs that the parses sees as one of: `:or`, `:and`, `:vector` or `:set`.

Should be easy to fix thou.

**Comment by ikitommi** [comment]
Checked all the wrapped specs, by adding the original spec reference to parse results, all of these work:

```clj
(s/def ::pos? (st/spec {:spec (partial pos?), :decode/string transform/string->long}))

(testing "referenced specs, #165"
  (is (= 1 (st/coerce (s/and ::pos?) "1" st/string-transformer)))
  (is (= 1 (st/coerce (s/or :default ::pos?) "1" st/string-transformer)))
  (is (= [1 2 3 :4] (st/coerce (s/coll-of ::pos?) ["1" "2" "3" :4] st/string-transformer)))
  (is (= {1 :2, :3 4} (st/coerce (s/map-of ::pos? ::pos?) {"1" :2, :3 "4"} st/string-transformer)))
  (is (= [1 :2 3 "4"] (st/coerce (s/tuple ::pos? keyword? ::pos?) ["1" "2" "3" "4"] st/string-transformer)))
  (is (= 1 (st/coerce (s/nilable ::pos?) "1" st/string-transformer))))
```

looks ok?

**Comment by mping** [comment]
@ikitommi looks really good!

**Comment by ikitommi** [comment]
FIxed in `0.9.0-alpha1`.
